exe/shell/main.go

```diff
@@ -179,25 +190,65 @@ func handleLine(line string) (out string, err error) {
 	}
 
 	if strings.HasPrefix(line, ":") {
-		args := strings.Split(line[1:], " ")
-		switch args[0] {
+		parts := strings.SplitN(line, " ", 2)
+		cmds := strings.Split(parts[0], ":")
+		cmds = cmds[1:]
+		switch cmds[0] {
+		case "help":
+			return cmdCmd()
 		case "address":
 			return addressCmd()
-		case "thread":
-			return threadCmd(args[1:])
 		case "threads":
 			return threadsCmd()
-		case "thread-address":
-			return threadAddressCmd()
-		case "add-follower":
-			return addFollowerCmd(args[1:])
-		default:
-			err = fmt.Errorf("unknown command: %s", args[0])
+		case "enter":
+			if len(parts) == 1 {
+				err = fmt.Errorf("missing thread name")
+				return
+			}
+			return enterCmd(parts[1])
+		case "exit":
+			threadID = thread.Undef
+			cursor = green(">  ")
 			return
+		case "add":
+			if len(parts) == 1 {
+				err = fmt.Errorf("missing thread name")
+				return
+			}
+			args := strings.Split(parts[1], " ")
+			return addCmd(args)
+		case "":
+			err = fmt.Errorf("missing command")
+			return
+		default:
+			var input string
+			if len(parts) > 1 {
+				input = parts[1]
+			}
+			return threadCmd(cmds, input)
 		}
 	}
 
-	err = sendMessage(line)
+	if threadID.Defined() {
+		err = sendMessage(threadID, line)
+	} else {
+		err = fmt.Errorf("enter a thread with `:enter` or specify thread name with :<name>")
+	}
+	return
+}
+
+func cmdCmd() (out string, err error) {
```

Comment:
I like the feel of the interactive cli you have here a lot... but kinda surprised there isn't an existing 'solution' to make generating things like help and usage msgs easier?

Reply:
There are some good shell libs but this is more a of a chat interface that allows for command. So, it's a bit different in feel. Mostly just playing around here though... when the time comes for a "real" shell, we can reevaluate.